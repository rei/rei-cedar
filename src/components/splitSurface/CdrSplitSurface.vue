<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import mapClasses from '../../utils/mapClasses';

/** Introduces a layered surface for distinct content sections, smoothing the transition from one page section to another */
defineOptions({
  name: 'CdrSplitSurface'
});

interface splitSurfaceProps {
  /** 
   * Sets the orientation of the surface to the top or bottom slot
   * @demoSelectMultiple false
   * @values top, bottom
   */
  surface?: 'top' | 'bottom'
  /** Sets the HTML tag for the split-surface element */
  tag?: string
}

const props = withDefaults(defineProps<splitSurfaceProps>(), {
  surface: 'bottom',
  tag: 'div'
});

const baseClass = 'cdr-split-surface'
const surfaceClass = computed(() => `${baseClass}--${props.surface}`);

const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="mapClasses(style, baseClass, surfaceClass)"
  >
    <div :class="style[`${baseClass}__top`]">
      <!-- @slot top slot -->
      <slot name="top" />
    </div>
    <div :class="style[`${baseClass}__bottom`]">
      <!-- @slot bottom slot -->
      <slot name="bottom" />
    </div>
  </component>
</template>

<style lang="scss" module src="./styles/CdrSplitSurface.module.scss">
</style>
