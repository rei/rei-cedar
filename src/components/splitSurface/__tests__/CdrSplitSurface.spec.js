import { shallowMount } from '../../../../test/vue-jest-style-workaround.js';
import CdrSplitSurface from '../CdrSplitSurface.vue';

describe('CdrSplitSurface', () => {
  it('default configuration', () => {
    const wrapper = shallowMount(CdrSplitSurface, {
      slots: {
        top: 'Top slot',
        bottom: 'Bottom slot'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('changes surface orientation', () => {
    const wrapper = shallowMount(CdrSplitSurface, {
      slots: {
        top: 'Top slot',
        bottom: 'Bottom slot'
      }
    });
    wrapper.setProps({
      surface: 'top'
    })
    expect(wrapper.element).toMatchSnapshot();
  });
});
