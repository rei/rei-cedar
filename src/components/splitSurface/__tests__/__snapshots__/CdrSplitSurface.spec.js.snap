// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrSplitSurface > changes surface orientation 1`] = `
<div
  class="cdr-split-surface cdr-split-surface--bottom"
>
  <div
    class="cdr-split-surface__top"
  >
    <!-- @slot top slot -->
    
    Top slot
    
  </div>
  <div
    class="cdr-split-surface__bottom"
  >
    <!-- @slot bottom slot -->
    
    Bottom slot
    
  </div>
</div>
`;

exports[`CdrSplitSurface > default configuration 1`] = `
<div
  class="cdr-split-surface cdr-split-surface--bottom"
>
  <div
    class="cdr-split-surface__top"
  >
    <!-- @slot top slot -->
    
    Top slot
    
  </div>
  <div
    class="cdr-split-surface__bottom"
  >
    <!-- @slot bottom slot -->
    
    Bottom slot
    
  </div>
</div>
`;
